lib/nerdtree/bookmark.vim

```diff
@@ -97,7 +97,7 @@ function! s:Bookmark.CacheBookmarks(silent)
 
                 let name = substitute(i, '^\(.\{-}\) .*$', '\1', '')
                 let path = substitute(i, '^.\{-} \(.*\)$', '\1', '')
-                let path = fnamemodify(path, ':p')
+                let path = fnamemodify(expand(path), ':p')
```

Comment:
@PhilRunninger: My issue with this change is that `expand()` does more than just expand environment variables. For example, if a target path contains wildcard characters, the bookmark will never be interpreted as valid.

A simple case of this is a file named something like `*file*.txt` in your home directory with other files named `file1.txt` and `file2.txt`, for example.

Calling...

```viml
:echo expand('$HOME/*file*.txt')
```

... will give you more than you asked for.  It is unlikely, but it *is* a bug, IMHO.

We need to expand *only* environment variables.  I'm not sure it can be done without a considerable amount of hacking.  You also have the problem of directory names with dollar signs in them.  What if one happens to match an environment variable (again, not likely)?

I recommend to leave this PR out, unless these problems are securely addressed.